New package: BusInjectionModel v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: BusInjectionModel
- Repository: https://github.com/NLaws/BusInjectionModel.jl
- Created by: @NLaws
- Version: v0.1.0
- Commit: 31344f1c66c868539cce6a5b33bd2d89a557d87f
- Reviewed by: @NLaws
- Reference: https://github.com/NLaws/BusInjectionModel.jl/commit/31344f1c66c868539cce6a5b33bd2d89a557d87f#commitcomment-150683574
- Description: Various forms of the Bus Injection Model (power flow equations) built on the CommonOPF.jl package
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6840317aa3862d9075e8fbe1b7f4b0372c0ada4c2111b72d482beda7bdcb7ce9afc47399df01db8236b42e5f0eb25414c6f7f3304a8e3eae6100f9abff56831993f50bb2688bf15860b8e78e1390a3ab2e0718bd048c8d338e76a28619fad3c53884458984ca62bec83dd785bd4e14d932dbe783ebb8c512069617fedd83a6162a087847e485366f6faf37c4bd4da50b05c9b3938a71a279350d7f66c26cfd76f5ebef3fc6e5998eb840a6a0354790d88 -->

goerz commented:
This package is currently empty and is thus [not ready for registration](https://github.com/JuliaRegistries/General?tab=readme-ov-file#should-i-register-my-package-now) yet. We do not allow for "placeholder" registration. If you are registering to make development workflows easier, please consider using a [LocalRegistry](https://github.com/GunnarFarneback/LocalRegistry.jl)

NLaws commented:
@goerz [noblock] I'm not sure what the issue is here: I think that the problem is
```julia
┌ Error: Was not able to successfully `Pkg.add` the package
└ @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/mNCS2/src/AutoMerge/guidelines.jl:836
┌ Info: Version can be `Pkg.add`ed
│   meets_this_guideline = false
└   message = "I was not able to install the package (i.e. `Pkg.add(\"BusInjectionModel\")` failed). See the AutoMerge logs for details."
```
But I also see:
```julia
[ Info: IMPORTANT: If you see any messages of the form "Error: Some registries failed to update"or "registry dirty", please disregard those messages. Those messages are normal and do not indicate an error.
```
and all the errors seem to be related to errors that the `IMPORTANT` message says to ignore.

goerz commented:
[noblock]

I don't see anything very obvious.

The logs to look at are in the "Checks" tab at the top of the page: https://github.com/JuliaRegistries/General/pull/115285/checks and specifically https://github.com/JuliaRegistries/General/actions/runs/11405524579/job/31737190398

The reported error is

```
ERROR: Error building `OpenDSSDirect`: 
ERROR: LoadError: ArgumentError: Package Downloads does not have NetworkOptions in its dependencies:
- You may have a partially installed environment. Try `Pkg.instantiate()`
  to ensure all packages in the environment are installed.
- Or, if you have Downloads checked out for development and have
  added NetworkOptions as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with Downloads
```

Seems a bit odd, but that would be the thing to investigate.

Other logs for the registration CI can be a bit misleading, and are generally for internal use only

This is what

```
[ Info: IMPORTANT: If you see any messages of the form "Error: Some registries failed to update"or "registry dirty", please disregard those messages. Those messages are normal and do not indicate an error.
```

is trying to tell you. Those "errors" are not what is causing the failure.


Usually, "I was not able to install/load the package" errors can be resolved by testing _locally_. In this case, I was able to clone and instantiate your package, so I (you) would have to look deeper.

I did notice that trying to run your test suite fails:

```
  Got exception outside of a @test
  OpenDSSDirect.Utils.OpenDSSDirectException("[ERROR 243] Redirect file not found: \"data/ieee13/IEEE13Nodeckt_no_trfxs.dss\"")
```

You definitely want to make sure that you can run your tests locally and on CI before registering. Note that red CI badge in your README. Make sure that is green, and then see if you are still getting errors when trying to register.

goerz commented:
> Note that red CI badge in your README

Oh, never mind, that's for the "pre" version of Julia. I recommend not routinely testing against pre-releases/nightlies of Julia as part of your regular CI. This will give you a lot of false positives. Instead, test against the oldest Julia version you support, and the current release.

goerz commented:
I should add, though: in order to be registered, your package will also need some documentation (in addition to passing the automatic checks). [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README

NLaws commented:
Thank you @goerz all very helpful! 

I found an issue in the OpenDSSDirect dependency that was causing things to fail. I'll come back to this with more documentation.

JuliaRegistrator committed:
New package: BusInjectionModel v0.1.0

UUID: 20cea600-32a0-4edf-8dad-2de078d2a706
Repo: https://github.com/NLaws/BusInjectionModel.jl.git
Tree: 5c10a035ca6792640eb1b6684da0c7966e6bae91

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

NLaws commented:
This PR can be closed in favor of https://github.com/JuliaRegistries/General/pull/122048
